### PR TITLE
Fixing issue #954

### PR DESCRIPTION
**Pull request by mikeparisstuff**
*Issue #, if available:*

#954

*Description of changes:*

This stops creating API keys when user pools auth is enabled.

Changes:
1. Moves the API key logic into the ModelAuthTransformer instead of the DynamoDBModelTransformer.
2. The CLI loads the auth transformer always.
3. Removed deprecated code that used to create a user pool from within the transform. This was only used by tests.
4. Updated tests such that they create their own user pool and pass it by id to CloudFormation.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graphql-transformers-e2e-tests/src/__tests__/ConnectionsWithAuthTests.e2e.test.ts [hunk 1]
```diff
@@ -232,46 +239,45 @@ beforeAll(async () => {
         console.log(`Using graphql url: ${GRAPHQL_ENDPOINT}`);
 
         // Get the details of the created user pool
-        const userPoolIdSelector = outputValueSelector(ResourceConstants.OUTPUTS.AuthCognitoUserPoolIdOutput)
-        const userPoolClientIdSelector = outputValueSelector(ResourceConstants.OUTPUTS.AuthCognitoUserPoolJSClientOutput)
-        const userPoolId = userPoolIdSelector(finishedStack.Outputs);
-        const userPoolClientId = userPoolClientIdSelector(finishedStack.Outputs);
-
+        // const userPoolIdSelector = outputValueSelector(ResourceConstants.OUTPUTS.AuthCognitoUserPoolIdOutput)
```
Comment (yuth) [inline packages/graphql-transformers-e2e-tests/src/__tests__/ConnectionsWithAuthTests.e2e.test.ts]: Can we get rid of this instead of commenting?

Comment (mikeparisstuff) [reply packages/graphql-transformers-e2e-tests/src/__tests__/ConnectionsWithAuthTests.e2e.test.ts]: yup

**Comment by mikeparisstuff** [comment]
Before merging this it would be nice to get this CLI integration tests for the api category wired up. I will work on this later this week.

File: packages/amplify-e2e-tests/__tests__/api.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    deleteProjectDir(projRoot);
+  });
+
+  it('init a project and add the simple_model api', async () => {
+    await initProjectWithProfile(projRoot, { name: 'simplemodel' });
+    const meta = getProjectMeta(projRoot).providers.awscloudformation;
+    expect(meta.Region).toBeDefined();
+    const { AuthRoleName, UnauthRoleName, UnauthRoleArn, AuthRoleArn, DeploymentBucketName } = meta;
+
+    await expect(UnauthRoleName).toBeIAMRoleWithArn(UnauthRoleArn);
```
Comment (yuth) [inline packages/amplify-e2e-tests/__tests__/api.test.ts]: Do we need these assertions?

File: packages/amplify-e2e-tests/__tests__/api.test.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    await expect(AuthRoleName).toBeIAMRoleWithArn(AuthRoleArn);
+    await expect(DeploymentBucketName).toBeAS3Bucket();
+
+    await addApiWithSimpleModel(projRoot, {});
+    await amplifyPush(projRoot);
+    const { output } = getProjectMeta(projRoot).api.simplemodel;
+    const { GraphQLAPIIdOutput, GraphQLAPIEndpointOutput, GraphQLAPIKeyOutput } = output;
+    
+    await expect(GraphQLAPIIdOutput).toBeDefined()
```
Comment (yuth) [inline packages/amplify-e2e-tests/__tests__/api.test.ts]: Can we assert that these values we have are indeed the right ones using aws-sdk

File: packages/amplify-e2e-tests/__tests__/api.test.ts (hunk 2 above)
Comment (mikeparisstuff) [reply packages/amplify-e2e-tests/__tests__/api.test.ts]: Removed these assertions.

File: packages/amplify-e2e-tests/__tests__/api.test.ts (hunk 3 above)
Comment (mikeparisstuff) [reply packages/amplify-e2e-tests/__tests__/api.test.ts]: Added a todo in the code for additional checking but I think pulling from the CloudFormation outputs should be enough for now.

**Comment by mikeparisstuff** [comment]
Thanks for the review @yuth. I am going to merge this.

**Pull request merged**